### PR TITLE
doc: update schema apply doc with sql examples

### PR DESCRIPTION
**Pull request by a8m**

### CODE REVIEW

**Review by rotemtam — APPROVED** [review]
💯

File: doc/md/declarative/apply.mdx [hunk 1]
```diff
@@ -11,10 +11,23 @@ With Atlas, users do not need to plan database schema changes themselves. Instea
 of figuring out the correct SQL statements to get their database to the desired state,
 Atlas supports a kind of workflow that we call _declarative schema migration_.
 With declarative schema migrations the user provides a connection string to the target database
-and the desired schema and Atlas does all of the planning.
+and the desired schema and Atlas does all the planning.
```
Comment (hilakashai) [inline doc/md/declarative/apply.mdx:14]:
```suggestion
and the desired schema, while Atlas does the planning.
```

File: doc/md/declarative/apply.mdx [hunk 2]
```diff
@@ -11,10 +11,23 @@ With Atlas, users do not need to plan database schema changes themselves. Instea
...
 
 [Read more about declarative workflows](/concepts/declarative-vs-versioned)
 
+## Flags
+
+The `schema apply` generates a migration plan and apply it to the database to bring it to the desired state.
```
Comment (hilakashai) [inline doc/md/declarative/apply.mdx:20]:
```suggestion
The `schema apply` command generates a migration plan and applies it to the database to bring it to the desired state.
```

File: doc/md/declarative/apply.mdx [hunk 3]
```diff
@@ -11,10 +11,23 @@ With Atlas, users do not need to plan database schema changes themselves. Instea
...
+A desired state can be specified using a database URL, HCL or SQL schema, or a migration directory.
```
Comment (hilakashai) [inline doc/md/declarative/apply.mdx:21]:
```suggestion
A desired state can be specified using a database URL, an HCL or SQL schema, or a migration directory.
```

Comment (hilakashai) [inline doc/md/declarative/apply.mdx:21]:
```suggestion
A desired state can be specified using a database URL, an HCL or SQL schema, or a migration directory.
```

File: doc/md/declarative/apply.mdx [hunk 4]
```diff
@@ -11,10 +11,23 @@ With Atlas, users do not need to plan database schema changes themselves. Instea
...
+
+* `--url` (`-u` accepted as well) - the [URL](/concepts/url) of database to be inspected.
```
Comment (hilakashai) [inline doc/md/declarative/apply.mdx:23]:
```suggestion
* `--url` (`-u` accepted as well) - the [URL](/concepts/url) of the database to be inspected.
```

File: doc/md/declarative/apply.mdx [hunk 5]
```diff
@@ -11,10 +11,23 @@ With Atlas, users do not need to plan database schema changes themselves. Instea
...
+* `--to` - a list of [URLs](/concepts/url) to the desired state: can be a database URL, HCL or
```
Comment (hilakashai) [inline doc/md/declarative/apply.mdx:24]:
```suggestion
* `--to` - a list of [URLs](/concepts/url) to the desired state: can be a database URL, an HCL or
```

Comment (hilakashai) [inline doc/md/declarative/apply.mdx:24]:
```suggestion
* `--to` - a list of [URLs](/concepts/url) to the desired state: can be a database URL, an HCL or
```

File: doc/md/declarative/apply.mdx [hunk 6]
```diff
@@ -39,11 +52,225 @@ This means in certain situations it may appear to Atlas as if some diff exists
...
+  --dev-url "sqlite://file?mode=memory"
+```
+
+</TabItem>
+</Tabs>
+
+
+### SQL schema
+
+The following example demonstrates how to use SQL schema file as the desired state and update the database schema to
```
Comment (hilakashai) [inline doc/md/declarative/apply.mdx:119]:
```suggestion
The following example demonstrates how to use a SQL schema file as the desired state and update the database schema to
```

File: doc/md/declarative/apply.mdx [hunk 7]
```diff
@@ -39,11 +52,225 @@ This means in certain situations it may appear to Atlas as if some diff exists
...
+  --dev-url "sqlite://file?mode=memory"
+```
+
+</TabItem>
+</Tabs>
+
+### Migrations directory
+
+The following example demonstrates how to use the migration directory as the desired state and update the database
+schema to match it. The URL for the migrations directory can contain two optional query parameters:
```
Comment (hilakashai) [inline doc/md/declarative/apply.mdx:175]:
```suggestion
schema to match it. The URL for the migration directory can contain two optional query parameters:
```

File: doc/md/declarative/apply.mdx [hunk 8]
```diff
@@ -39,11 +52,225 @@ This means in certain situations it may appear to Atlas as if some diff exists
...
+* `format` - migrations directory format: atlas (default), golang-migrate, goose, dbmate, flyway, liquibase.
```
Comment (hilakashai) [inline doc/md/declarative/apply.mdx:176]:
```suggestion
* `format` - migration directory format: atlas (default), golang-migrate, goose, dbmate, flyway, liquibase.
```

File: doc/md/declarative/apply.mdx [hunk 9]
```diff
@@ -39,11 +52,225 @@ This means in certain situations it may appear to Atlas as if some diff exists
...
+* `version` - until which version to read migration files. By default, all migrations are read.
```
Comment (hilakashai) [inline doc/md/declarative/apply.mdx:177]:
```suggestion
* `version` - until which version of migration files to read. By default, all migrations are read.
```

File: doc/md/declarative/apply.mdx [hunk 10]
```diff
@@ -39,11 +52,225 @@ This means in certain situations it may appear to Atlas as if some diff exists
...
+  --dev-url "sqlite://file?mode=memory"
+```
+
+</TabItem>
+</Tabs>
+
+### Down migrations
+
+Migrations that "roll back" or reverse changes made to the database schema are called "down migrations". These are often
+used during local development to undo the changes made by corresponding "up migrations." Atlas, follows a linear
```
Comment (hilakashai) [inline doc/md/declarative/apply.mdx:232]:
```suggestion
used during local development to undo the changes made by corresponding "up migrations". Atlas follows a linear
```

Comment (hilakashai) [inline doc/md/declarative/apply.mdx:232]:
```suggestion
used during local development to undo the changes made by corresponding "up migrations". Atlas follows a linear
```

File: doc/md/declarative/apply.mdx [hunk 11]
```diff
@@ -39,11 +52,225 @@ This means in certain situations it may appear to Atlas as if some diff exists
...
+migration history model, in which all migration files are "roll-forward". However, it is still possible to clean or
+revert schema changes made by specific migration files using the `schema clean` or `schema apply` commands. For example:
+
+1\. If a migration file(s) was already deleted, you can use the [command above](#migration-directory) to migrate the
+database state to the state defined by the migrations directory. For example:
```
Comment (hilakashai) [inline doc/md/declarative/apply.mdx:237]:
```suggestion
database state to the state defined by the migration directory. For example:
```

Comment (hilakashai) [inline doc/md/declarative/apply.mdx:237]:
```suggestion
database state to the state defined by the migration directory. For example:
```

File: doc/md/declarative/apply.mdx [hunk 12]
```diff
@@ -39,11 +52,225 @@ This means in certain situations it may appear to Atlas as if some diff exists
...
+  --dev-url "docker://mysql/8/example"
+```
+
+```shell title="tree ." {6}
+.
+└── migrations
+    ├── 20220925092817_initial.sql
+    ├── 20220925094021_second.sql
+    ├── 20220925094437_third.sql
+    ├── 20220925094438_forth.sql # changes made by this file are reverted.
```
Comment (hilakashai) [inline doc/md/declarative/apply.mdx:262]:
```suggestion
    ├── 20220925094438_fourth.sql # changes made by this file are reverted.
```

File: doc/md/declarative/inspect.mdx [hunk 13]
```diff
@@ -21,10 +21,10 @@ language that they can use as the starting point for working with this database.
 
 When using `schema inspect` to inspect an existing database, users may supply multiple
 parameters:
-* `--url` (required, `-u` accepted as well) - the [URL](/concepts/url) of database to be inspected.
+* `--url` (`-u` accepted as well) - the [URL](/concepts/url) of database to be inspected.
```
Comment (hilakashai) [inline doc/md/declarative/inspect.mdx:24]:
```suggestion
* `--url` (`-u` accepted as well) - the [URL](/concepts/url) of the database to be inspected.
```

**Pull request merged**